pages/articles/SectionsHandler.inc.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/**
+ * @file pages/articles/SectionsHandler.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2003-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class SectionsHandler
+ * @ingroup pages_articles
+ *
+ * @brief Handle requests for sections functions.
+ *
+ */
+
+use APP\core\Application;
+use APP\facades\Repo;
+use APP\handler\Handler;
+use APP\security\authorization\OjsJournalMustPublishPolicy;
+use APP\submission\Submission;
+use APP\template\TemplateManager;
+use PKP\db\DAORegistry;
+use PKP\security\authorization\ContextRequiredPolicy;
+
+class SectionsHandler extends Handler
+{
+    /** sections associated with the request **/
+    public $sections;
+
+    /**
+     * @copydoc PKPHandler::authorize()
+     */
+    public function authorize($request, &$args, $roleAssignments)
+    {
+        $this->addPolicy(new ContextRequiredPolicy($request));
+        $this->addPolicy(new OjsJournalMustPublishPolicy($request));
+        return parent::authorize($request, $args, $roleAssignments);
+    }
```

Comment:
The `SectionsHandler` also needs to handle the `index` page, which should show a list of the sections in the journal with their description and links to their browsing pages (and maybe a count of total entries).

Reply:
ok, sounds good!

But what about CategoryHandler? I do not think we have something like that there either?

Reply:
hmm, just thinking that if we want to have paths like
```
articles/section/index
articles/category/index
```

We can not handle index as `$op` here https://github.com/pkp/ojs/blob/40bbc989aa0ff2ec2747cf5396f908baa3e1fc9c/pages/articles/index.php#L19

Instead we need see what the `$sectionUrlPath` is and handle the call based on that. This would also mean that we would maybe need to block the usage of `index` as a section path.

Or is there another way of handling this? I looked around the other page handlers but did not find a similar case.

Reply:
Yeah I'm not sure. But I'd say to be safe we should block `index` as a section path, article `urlPath`, category path, etc.